scenarios.md

```diff
@@ -199,6 +205,63 @@ A deployment requires a mydb image. The mydb image is routinely updated for secu
+
+In this scenario, the attacker didn't have the private keys to sign the new content. They pushed unsigned content to existing tags.
+
+1. An attacker gains access to the credentials, allowing them to push new content.
+1. The attacker pushes an artifact with compatible behavior, with an additional exploit.
+1. The new compromised artifact is pushed with an existing tag, but not signed.
+1. Consumers of the artifact that don't check for signatures are unaware of the exploit as it produces the same original behavior.
+1. Historical data is maintained, enabling forensics when it's realized compromised content was pushed.
+1. All newly pushed content, from the time of compromise is removed, restoring content and tags to their previous state.
+1. Consumers that pulled the new exploited tags will identify the digest are different, pulling the restored/original manifest and its associated content.
+
+**Implications of this requirement:**
+
+1. The potential damage/risk to consumers must be limited.
+1. There must be a secure way for users to recover to a known, secure state and verify this has occurred even in the face of an attacker that can act as a man-in-the-middle on the network.
+1. How a registry implements the rollback to the previously secured state is a differentiating capability. The scenario simply calls out a compromise that must be discoverable through forensic logging of who, when and what was pushed. The logged updates must include previous digests that represented updated tags allowing a registry, or external tools, to reset its original state.
+1. A registry may choose to make repos, or the entire registry, restricted to pushing only signed content, and potentially only signed content with one or more keys. Notary v2 doesn't require this capability, rather highlights the scenario for registry operators to innovate means to secure from this scenario.
+
+### Scenario #8: A Developer Discloses Their Key and/or Credentials
+
+A developer accidentally discloses the private key they use to certify their software is authentic. 
+
+1. A developer accidentally checks their private key into a github repository.
+1. A developer references a compromised package that searches for private keys within the build system and `curl`s it a location of an attacker.
+1. A developer may have scripted the means by which secured content is added to a registry and the script (and related credentials & secrets) have all been disclosed.
+1. Whether 5 minutes or 5 days, the developer must assume the pubic key was copied and new content may be pushed to this, or any other registry claiming it represents the identify of the private key.
+1. The attacker may not have credentials to push content to a the registry in focus, but they may push similarly named content to other public registries, misrepresenting the identity.
+1. The developer revokes their public key, indicating it's no longer valid.
+1. The developer issues a new public key and re-signs their content.
```

Comment:
Forgive me if this is covered elsewhere, I'm quite new here -- what are the implications, and verifications, needed for this new key authenticity? I don't see it in the implications below.

Reply:
I'm leaving this open to see if we cover this in the key management working group.